src/main/java/com/avispl/symphony/dal/communicator/middleatlantic/select/MiddleAtlanticPowerUnitCommunicator.java

```diff
@@ -0,0 +1,626 @@
+                String.valueOf(PREVIOUS_SUBCMD_INVALID), String.valueOf(PREVIOUS_BYTE_COUNT_INVALID),
+                String.valueOf(DATA_BYTES_INVALID), String.valueOf(BAD_CREDS),
+                String.valueOf(UNKNOWN), String.valueOf(ACCESS_DENIED)));
+
+        this.setCommandSuccessList(Collections.singletonList("\n"));
+    }
+
+    @Override
+    public void controlProperty(ControllableProperty controllableProperty) throws Exception {
+        String property = controllableProperty.getProperty();
+        String value = String.valueOf(controllableProperty.getValue());
+
+        if(logger.isInfoEnabled()){
+            logger.debug(String.format("Received controllable property '%s' with value '%s'", property, value));
+        }
+        refreshTCPSession();
+        controlOperationsLock.lock();
+        try {
+            if(property.equals(SEQUENCE_DOWN_COMMAND_NAME)){
+                boolean result = sendSequencingCommand(buildSequenceCommand(DOWN_SEQUENCE));
+                if(result) {
+                    updateLocalSequencingStatistics(DOWN_SEQUENCE);
+                }
+            } else if (property.equals(SEQUENCE_UP_COMMAND_NAME)){
+                boolean result = sendSequencingCommand(buildSequenceCommand(UP_SEQUENCE));
+                if(result){
+                    updateLocalSequencingStatistics(UP_SEQUENCE);
+                }
+            } else if(property.matches(".+?-\\s\\d")){
+                byte outletNumber = Byte.parseByte(property.substring(property.lastIndexOf(" ") + 1));
+                byte response;
+                switch (value) {
+                    case "1":
+                        response = sendControlCommand(buildWriteOutletStatusCommandBytes(outletNumber, (byte) 0x01));
+                        updateLocalOutletState(property, "1", String.valueOf(response));
+                        break;
+                    case "0":
+                        response = sendControlCommand(buildWriteOutletStatusCommandBytes(outletNumber, (byte) 0x00));
+                        updateLocalOutletState(property, "0", String.valueOf(response));
+                        break;
+                    default:
+                        break;
+                }
+            }
+        } finally {
+            controlOperationsLock.unlock();
+        }
+    }
+
+    @Override
+    public void controlProperties(List<ControllableProperty> controllablePropertyList) throws Exception {
+        if (CollectionUtils.isEmpty(controllablePropertyList)) {
+            throw new IllegalArgumentException("NetGearCommunicator: Controllable properties cannot be null or empty");
+        }
+
+        for(ControllableProperty controllableProperty: controllablePropertyList){
+            controlProperty(controllableProperty);
+        }
+    }
+
+    @Override
+    public List<Statistics> getMultipleStatistics() throws Exception {
+        ExtendedStatistics statistics = new ExtendedStatistics();
```

Comment:
Not thread safe - localStatistics is not protected by lock here while in all other places it is protected.